pipelined/bevy_render2/src/lib.rs

```diff
@@ -178,6 +180,12 @@ impl Plugin for RenderPlugin {
 }
 
 fn extract(app_world: &mut World, render_app: &mut App) {
+    // Update the render world clear color resource if it was added or changed in the app world
+    if app_world.is_resource_added::<ClearColor>() || app_world.is_resource_changed::<ClearColor>()
```

Comment:
I don't think this should be updated here / hard coded in the "extract step execution logic. I think ClearColor should probably live in `bevy_core_pipeline` as it is behavior specific to "bevy's core render pipeline", not core renderer logic. It should just be another "extract stage system". That should also resolve the change tracking issue.

You chould only need to query for Changed. "added" things also flip the "changed" flag.

Reply:
OK, cool, that makes sense. I'll update as soon as I get back to this. :+1:

Reply:
Fixed it. I like that much better!

I'm really liking the extract →prepare →queue rendering model. Very nice.